request/multiple_kernels.rst

```diff
@@ -0,0 +1,149 @@
+.. index::
+   single: kernel, performance
+
+How To Create Symfony Applications with Multiple Kernels
+========================================================
+
+In most Symfony applications, incoming requests are processed by the
+``web/app.php`` front controller, which instantiates the ``app/AppKernel.php``
+class to create the application kernel that loads the bundles and generates the
+response.
+
+This single kernel approach is a convenient default provided by the Symfony
+Standard edition, but Symfony applications can define any number of kernels.
+Whereas :doc:`environments </configuration/environments>` execute the same
+application with different configurations, kernels can execute different parts
+of the same application.
+
+These are some of the common use cases for creating multiple kernels:
+
+* An application that defines an API could define two kernels for performance
+  reasons. The first kernel would serve the regular application and the second
+  one would only respond to the API requests, loading less bundles and enabling
+  less features;
+* A highly sensitive application could define two kernels. The first one would
+  only load the routes that match the parts of the application exposed to the
+  public. The second kernel would load the rest of the application and its
+  access would be protected by the web server;
+* An application that uses a bundle which doesn't allow multiple instances could
```

Comment:
OK. I'm probably going to remove this use case. I added because the other day one person asked me if EasyAdmin supported defining two different backends in the same app. The bundle itself doesn't support that ... but with this multi-kernel trick, everything is easily solved.

Can anyone think of other use cases or do we just leave the first two proposed ones? Thanks.